e2e/test/scenarios/dashboard-filters/reproductions/12720-no-data-permissions-connected-filter.cy.spec.js

```diff
@@ -42,53 +47,39 @@ describe("issue 12720", () => {
 
     cy.createNativeQuestion(questionDetails).then(
       ({ body: { id: SQL_ID } }) => {
-        cy.request("POST", "/api/dashboard/1/cards", {
-          cardId: SQL_ID,
-          row: 0,
-          col: 6, // making sure it doesn't overlap the existing card
-          size_x: 5,
-          size_y: 5,
-        }).then(({ body: { id: SQL_DASH_CARD_ID } }) => {
-          cy.log(
-            "Edit both cards (adjust their size and connect them to the filter)",
-          );
-
-          cy.request("PUT", "/api/dashboard/1/cards", {
-            cards: [
-              {
-                id: 1,
-                card_id: 1,
-                row: 0,
-                col: 0,
-                size_x: 5,
-                size_y: 5,
-                parameter_mappings: [
-                  {
-                    parameter_id: dashboardFilter.id,
-                    card_id: 1,
-                    target: ["dimension", ["field", ORDERS.CREATED_AT, null]],
-                  },
-                ],
-                visualization_settings: {},
-              },
-              {
-                id: SQL_DASH_CARD_ID,
-                card_id: SQL_ID,
-                row: 0,
-                col: 6,
-                size_x: 5,
-                size_y: 5,
-                parameter_mappings: [
-                  {
-                    parameter_id: dashboardFilter.id,
-                    card_id: SQL_ID,
-                    target: ["dimension", ["template-tag", "filter"]],
-                  },
-                ],
-                visualization_settings: {},
-              },
-            ],
-          });
+        addCardsToDashboard({
+          dashboard_id: 1,
+          cards: [
+            {
+              card_id: SQL_ID,
+              row: 0,
+              col: 6, // making sure it doesn't overlap the existing card
+              size_x: 5,
+              size_y: 5,
+              parameter_mappings: [
+                {
+                  parameter_id: dashboardFilter.id,
+                  card_id: SQL_ID,
+                  target: ["dimension", ["template-tag", "filter"]],
+                },
+              ],
+            },
+            // add filter to existing card
+            {
+              card_id: 1,
+              row: 0,
+              col: 0,
+              size_x: 5,
+              size_y: 5,
+              parameter_mappings: [
+                {
+                  parameter_id: dashboardFilter.id,
+                  card_id: 1,
+                  target: ["dimension", ["field", ORDERS.CREATED_AT, null]],
+                },
+              ],
+            },
+          ],
```

Comment:
The new test is actually creating one new card and updating the existing one. In the `PUT` request that's being sent, the dashcard with `card_id` `SQL_ID` is being created, since it was not on the dashboard already, while the card with `card_id` `1` is being updated, since it already was on the dashboard. 

This is probably unclear due to the name of the method, perhaps it should be renamed to `addOrUpdateDashCards`?